src/web3py/middleware.py

```diff
@@ -29,31 +29,31 @@ def middleware(method: RPCEndpoint, params: Any) -> RPCResponse:
         except:
             domain = 'unavailable'
 
-        try:
-            with ETH1_RPC_REQUESTS_DURATION.time():
+        with EL_REQUESTS_DURATION.time() as t:
+            try:
                 response = make_request(method, params)
-        except HTTPError as ex:
-            failed: Response = ex.response
-            ETH1_RPC_REQUESTS.labels(
-                method=method,
-                code=failed.status_code,
+            except HTTPError as ex:
+                failed: Response = ex.response
+                t.labels(
+                    method=method,
+                    code=failed.status_code,
+                    domain=domain,
+                )
+                raise
+
+            # https://www.jsonrpc.org/specification#error_object
+            # https://eth.wiki/json-rpc/json-rpc-error-codes-improvement-proposal
+            error = response.get("error")
+            code: int = 0
+            if isinstance(error, dict):
+                code = error.get("code") or code
+
+            t.labels(
+                name=method,
```

Comment:
99% of method_names would be eth_call.
Lets pass here some of params like "to" (address) and "method" method in bytes or name (if it could be done)

Reply:
fixed